docs/framework/20categories/30implementation/upgrade.md

```diff
@@ -1,13 +1,12 @@
 ### Upgradable
-TODO: This needs some thought.
+Like any software, smart contracts software can have bugs. This is true of both code that appears to be extremely simple and more complex code. Additionally, applications may require new features. As such, many projects have the ability to upgrade their smart contract code.
 
-Upgrading can lead to rug-pulling, possibly by attackers who gain access to the address with upgrade privileges.
-However, code often has bugs. Additionally, new features can be added. Hence, the ability to upgrade is important.
-Are different upgrade mechanisms more and less risky?
+Having the ability to upgrade a contract is inherently risky as it can lead to rug-pulling attacks, where the operators of a project change the contract logic and steal customer funds (for instance the [Hunter Defi](https://twitter.com/CertiKAlert/status/1523153063875575809) rug-pull). However, as a proportion of both the number of cypto rug pull attacks and as a proportion of the projects that use upgradable contracts, using contract upgrade as a method of stealing funds is very rare. We have no hard statistics on this; this is just an observation.
```

Comment:
I think another reason why upgrades could be a source of risk is that they can introduce new vulnerabilities, either because of logic bugs in the new code or because of how that code interacts with the existing state (e.g. Nomad)

Reply:
True.
Adding text to discuss this.